OPHD/MicroPather/micropather.cpp

```diff
@@ -848,9 +848,10 @@ void MicroPather::GetCacheData(CacheData* data)
 
 		data->hit = pathCache->hit;
 		data->miss = pathCache->miss;
-		if (data->hit + data->miss)
+		const auto hitAsFloat = static_cast<float>(data->hit);
+		if (const float total = hitAsFloat + data->miss)

```

Comment:
This still relies on implicit conversion of one operand. I find this style very unintuitive. I think it makes more sense to either case both operands, or do the addition first, and then cast the result.

----

Second thought, if we use `float` rather than double, there is no increase in size to warn about.

----

Consider:
```cpp
auto total = data->hit + data->miss;
if (total)
{
  data->hitFraction = static_cast<float>(data->hit) / static_cast<float>(total);
}
```

In regards to testing an `int` in the `if` while using a `float` in the calculation, now that I think about it, it should be impossible for a non-zero integer to convert to a floating point 0. In the upper ranges, it may lose precision, but it would be a less precise very large number, not zero. There would be no overflow or wraparound during the conversion, only rounding to less precise large values. That would happen at about 2^24.